src/Infrastructure/TimeMgr/src/ESMCI_Time.C

```diff
@@ -830,13 +830,48 @@ namespace ESMCI{
       return(rc);
     }
 
-    if (this->calendar->calkindflag == ESMC_CALKIND_JULIANDAY ||
-        this->calendar->calkindflag == ESMC_CALKIND_MODJULIANDAY ||
-        this->calendar->calkindflag == ESMC_CALKIND_NOCALENDAR) {
-      ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
-        "; calkindflag is JULIANDAY, "
-        "MODJULIANDAY or NOCALENDAR.", ESMC_CONTEXT, &rc);
-      return(rc);
+    switch (this->calendar->calkindflag)
+    {
+        case ESMC_CALKIND_NOLEAP:
+          ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
+            "; calkindflag is NOLEAP. Only GREGORIAN and JULIAN are supported.",
+            ESMC_CONTEXT, &rc);
+          return(rc);
+          break;
+
+        case ESMC_CALKIND_360DAY:
+          ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
+            "; calkindflag is 360DAY. Only GREGORIAN and JULIAN are supported.",
+            ESMC_CONTEXT, &rc);
+          break;
```

Comment:
Hi Ufuk, I noticed that there wasn't a return here and I think that there should be, would you please confirm that. Thanks!

Reply:
@oehmke Good catch. Thanks. I add it.